src/polynomial.cpp

```diff
@@ -0,0 +1,310 @@
+#include "polynomial.h"
+#include "add.h"
+#include "mul.h"
+#include "pow.h"
+#include "constants.h"
+
+namespace SymEngine {
+
+    Polynomial::Polynomial(const RCP<const Symbol> &var, map_uint_mpz&& dict) :
```

Comment:
Yes, compared to the rest of the classes, most of the functions here should be shifted by 4 spaces. Shall I do that?